### PR TITLE
Setup deployment

### LINKED ISSUE #20: Run on OKD
I don't know all the steps, I am guessing at the minimum we will probably need: 
- a dockerfile 
- api calls to go to an actual url (not localhost), probably settable by some env variable?

### PR DESCRIPTION
**Pull request by mxmeinhold**
Resolves #20 
- Adds a Dockerfile
- Adds docker build to travis
- Includes a couple of fixes for issues we ran into when getting it running.
- Adds all the deploy yaml (except the config map).
  - The config map needs at minimum DB_URL

TODOs I see from here:
- It doesn't pull from env for some things it seems, so the config map doesn't work, and we had to use .env
- There's no archive page
- There's no 'this many people still need to vote' or sum of votes so far
- The test poll always shows up on startup (I dunno I think this is a feature)
- The build refers to this branch (`docker`), and there's no dev setup.
- There's no favicon
- The dashboard and setting links in the user dropdown don't do anything
- Pulling from/integrating with conditional to auto setup polls would be neat.
- I'd like to get the docker image building with nvm, or at least cleanup the dockerfile.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+RUN source $NVM_DIR/nvm.sh; \
+    nvm use \
+    && npm install --production 
+
+COPY . /opt/vote
+
+RUN source $NVM_DIR/nvm.sh; \
+    nvm use \
+    && npm run build
```
Comment (devinmatte) [inline Dockerfile:48]: We don't currently have a build script

**Comment by mxmeinhold** [comment]
@RebeccaSyria Thoughts on getting this merged in? I think we should probably merge this and tag it as 1.0.

A couple commits are kinda messy, but without rewriting history on the whole branch that's not gonna get cleaned up, so I'd sign off on merging this.

@devinmatte @galenguyer @WillNilges If any of you feel like a review, go wild, but I don't think it's _strictly_ necessary.

**Comment by WillNilges** [comment]
@mxmeinhold That was my thought on this. I say go for it if you're cool with it. I think we've demonstrated that it works, and 1.0 works. Now we gotta focus on a 1.1. Do we have a project board yet? Might wanna start one.

**Review by WillNilges — APPROVED** [review]
:mailbox_with_mail:

(for good measure)

**Review by RebeccaSyria — APPROVED** [review]

**Comment by mxmeinhold** [comment]
@WillNilges I don't think we do, but that's mostly a matter of tagging issues with milestones.

**Comment by mxmeinhold** [comment]
I'll give a little time in case @galenguyer or @devinmatte want to look at this, but otherwise I'll plan to merge this afternoon.

**Review by devinmatte — APPROVED** [review]
#yoloops

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,52 @@
+FROM node:12-buster-slim
+LABEL maintainer="Max Meinhold <mxmeinhold@gmail.com>"
+
+EXPOSE 8080
+
+ENV NODE_ENV production
+
+RUN mkdir /opt/vote
```
Comment (galenguyer) [inline Dockerfile:8]: WORKDIR creates the directory, no need to mkdir

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+#ENV PATH $NVM_DIR/versions/node/v$NODE_VERSION/bin:$PATH
+
+COPY package.json ./
+COPY client/package.json ./client/
+
+#RUN source $NVM_DIR/nvm.sh; \
+#    nvm use \
+#    && npm install --production=false \
+#    && cd client && npm install --production=false
+RUN npm install --production=false && cd client && npm install --production=false
```
Comment (galenguyer) [inline Dockerfile:40]: If you have a package.lock file, copy it here and use npm ci

**Comment by mxmeinhold** [comment]
Oh we also have to change the build to look at prod before deleting this, will fix that

File: Dockerfile (hunk 3 above)
Comment (mxmeinhold) [reply Dockerfile:40]: We don't have a package lock.

**Pull request merged**